FinalEngine.Editor.ViewModels/Docking/Tools/Inspectors/PropertiesToolViewModel.cs

```diff
@@ -14,22 +17,71 @@ namespace FinalEngine.Editor.ViewModels.Docking.Tools.Inspectors;
 /// <seealso cref="IPropertiesToolViewModel" />
 public sealed class PropertiesToolViewModel : ToolViewModelBase, IPropertiesToolViewModel
 {
+    /// <summary>
+    /// The messenger.
+    /// </summary>
+    private readonly IMessenger messenger;
+
+    /// <summary>
+    /// The current view model to be shown in the properties view.
+    /// </summary>
+    private object? currentViewModel;
+
     /// <summary>
     /// Initializes a new instance of the <see cref="PropertiesToolViewModel"/> class.
     /// </summary>
     /// <param name="logger">
     /// The logger.
     /// </param>
-    public PropertiesToolViewModel(ILogger<PropertiesToolViewModel> logger)
+    /// <param name="messenger">
+    /// The messenger.
+    /// </param>
+    public PropertiesToolViewModel(
+        ILogger<PropertiesToolViewModel> logger,
+        IMessenger messenger)
     {
         if (logger == null)
         {
             throw new ArgumentNullException(nameof(logger));
         }
 
+        this.messenger = messenger ?? throw new ArgumentNullException(nameof(messenger));
+
         this.Title = "Properties";
         this.ContentID = "Properties";
 
         logger.LogInformation($"Initializing {this.Title}...");
+
+        this.messenger.Register<EntitySelectedMessage>(this, this.HandleEntitySelected);
+    }
+
+    /// <inheritdoc/>
+    public object? CurrentViewModel
+    {
+        get { return this.currentViewModel; }
+        private set { this.SetProperty(ref this.currentViewModel, value); }
+    }
+
+    /// <summary>
+    /// Handles the <see cref="EntitySelectedMessage"/> and updates the view.
+    /// </summary>
+    /// <param name="recipient">
+    /// The recipient.
+    /// </param>
+    /// <param name="message">
+    /// The message.
+    /// </param>
+    /// <exception cref="System.ArgumentNullException">
+    /// The specified <paramref name="message"/> parameter cannot be null.
+    /// </exception>
+    private void HandleEntitySelected(object recipient, EntitySelectedMessage message)
+    {
+        if (message == null)
+        {
+            throw new ArgumentNullException(nameof(message));
+        }
+
+        this.Title = "Entity Inspector";
+        this.CurrentViewModel = new EntityComponentsViewModel();
```

Comment:
Use `IFactory<IEntityComponentViewMoedl>` and just set an `Entity` property once it's created.

Reply:
Let's chuck this in the backlog and worry about it if its required.